src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2177,6 +2185,157 @@ def import_disk_msdosfs_locales(self):
+                        )
+
+                        logger.warning("It will be obligatory to notify GELI that the provider has been resized: %r",
+                                       join_commandline(geli_resize_cmd))
+                        logger.warning("Or to resize provider back: %r",
+                                       join_commandline(rollback_cmd))
+
+                        geli_resize.append((geli_resize_cmd, rollback_cmd))
+            finally:
+                if geli_resize:
+                    failed_rollback = []
+
+                    try:
+                        lock_job = await self.middleware.call("pool.lock", pool["id"], options["geli"]["passphrase"])
+                        await lock_job.wait()
+                    except Exception:
+                        logger.warning("Error locking pool", exc_info=True)
+
+                        for geli_resize_cmd, rollback_cmd in geli_resize:
+                            if not await self.__run_rollback_cmd(rollback_cmd):
+                                failed_rollback.append(rollback_cmd)
+
+                        if failed_rollback:
+                            raise CallError("Locking your encrypted pool failed and rolling back changes failed too. "
+                                            "You'll need to run the following commands manually:\n%s",
+                                            "\n".join(map(join_commandline, failed_rollback)))
+                    else:
+                        for geli_resize_cmd, rollback_cmd in geli_resize:
+                            try:
+                                await run(*geli_resize_cmd, encoding="utf-8", errors="ignore")
+                            except subprocess.CalledProcessError as geli_resize_error:
+                                if geli_resize_error.stderr.strip() == "geli: Size hasn't changed.":
+                                    logger.info("%s: %s",
+                                                join_commandline(geli_resize_cmd),
+                                                geli_resize_error.stderr.strip())
+                                else:
+                                    logger.error("%r failed: %s. Resizing partition back",
+                                                 join_commandline(geli_resize_cmd),
+                                                 geli_resize_error.stderr.strip())
+                                    if not await self.__run_rollback_cmd(rollback_cmd):
+                                        failed_rollback.append(rollback_cmd)
+
+                        if failed_rollback:
+                            raise CallError("Resizing partitions of your encrypted pool failed and rolling back "
+                                            "changes failed too. You'll need to run the following commands manually:\n"
+                                            "%s",
+                                            "\n".join(map(join_commandline, failed_rollback)))
```

Comment:
Dont we already show the command in __run_rollback_cmd ?

Reply:
`__run_rollback_cmd` logs it to middleware log, here we output it to user.